learn/testing-ballerina-code/writing-tests.md

```diff
@@ -0,0 +1,348 @@
+  - /learn/testing-ballerina-code/writing-tests
+---
+
+# Writing Tests
+
+## Project Structure
+
+
+```
+project-name/
+  Ballerina.toml
+    src/
+      module1/
+        main.bal
+        Module.md 
+        [resources/]   
+        tests/       	  # Module-specific tests
+          main_test.bal   # The test file for main.bal
+          [resources]	  # Resources for the tests
+```
+
+
+
+## Test source files
+
+Unit tests bound to a module need to be placed in a subfolder called ***tests/*** within the module. In a standard
+ Ballerina project, a module is mapped to a test suite. All tests within a module’s *tests/* subfolder are
+  considered to be part of the same test suite.
+
+The test source files could have any name. The test functions are just Ballerina functions that use a special
+ annotation to mark the function as a test. Test functions must be specified with the `@test:Config {}` annotation and there is no restriction on the test function name.
+
+
+## Test Resources
+
+The resources subdirectory found within the *tests/* directory is meant to contain any files or resources that are
+ exclusively required for testing. You can access the resource files either using the absolute path or using the path relative to the project root.
+
+## Defining a test
+
+The test module provides the necessary annotations to construct a test suite. Therefore, importing the test module is essential in order to write Ballerina tests.
+
+
+```
+import ballerina/test;
+```
+
+
+Once the test module is imported, the following annotation can be used to write a test function.
+
+
+### @test:Config {}
+
+The function specified following the annotation is a test function. This annotation supports the following value fields.
+
+
+*   ***enable: {true&#124;false}*** - Enable/disable the test. Default: true
+*   ***before: "&lt;function name&gt;"*** - Name of the function to be run just before the test is run. Default: none
+*   ***after: "&lt;function name&gt;"*** - Name of the function to be run just after the test is run.
+*   ***dependsOn: ["&lt;function names>", …]*** - List of function names on which the test function depends. The
+ order in
+ which the comma-separated list appears has no prominence. In case there needs to be an order, define a sequence of test functions with one pointing to another based on the dependencies using the `dependsOn` parameter in each one’s config in order to control the order of test execution.
+*   ***dataProvider: “&lt;function name>”*** - Specifies the name of the function that will be used to provide the value
+ sets to execute the test against.
+*   ***groups: [“&lt;test group name”, …]*** - A comma-separated list of test group names (one or more) that this test
+ belongs to.
+
+***Example:***
+
+
+```ballerina
+import ballerina/io;
+import ballerina/test;
+
+function beforeFunc() {
+    // This is the before Test Function
+}
+
+function afterFunc() {
+    // This is the before Test Function
```

Comment:
```suggestion
    // This is the before Test Function
```
Same comment as above repeated here. Please check.

Reply:
Corrected it to `after Test function.`